pkg/agent/openflow/pipeline.go

```diff
@@ -353,6 +356,15 @@ const (
 
 	// EtherTypeDot1q is used when adding 802.1Q VLAN header in OVS action
 	EtherTypeDot1q = 0x8100
+
+	// dsrServiceSourcePortLearnedBits represents the bits of source port the flows learned for DSR Service will match.
+	// Using 8 bits mean it will generate at most 256 learned flows for this Service.
+	// For example, packets whose source port is 10000 will share the learned flow with packets whose source port is 10256.
+	dsrServiceSourcePortLearnedBits = 8
+	// dsrServiceConnectionIdleTimeout represents the idle timeout of the flows learned for DSR Service.
+	// 160 means the learned flows will be deleted if the flow is not used by any connections in 160s.
+	// It tolerates 1 keep-alive drop (net.ipv4.tcp_keepalive_intvl defaults to 75) and a deviation of 10s for long connections.
+	dsrServiceConnectionIdleTimeout = 160
```

Comment:
quick question: if my Service has SessionAffinity, will enabling DSR for the Service "break" SessionAffinity? In other words, if a given client opens one short TCP connection to the Service, then waits 3 minutes, then opens a second short TCP connection to the Service, could the second connection be forwarded to a different backend Node (assuming some changes in the cluster, such as new Service endpoints)?

Reply:
Ok, that sounds good to me.

So to recap: we are being very conservative with this first version of the DSR implementation. Nat mode and internal service traffic in DSR mode are not impacted at all. The implementation uses:
- dp_hash for backend Node selection
- one learned flow per connection (with FIN timeout), which means that we can have up to one learned flow per active connection

DSR will improve performance, but is not recommended for services with a very large number of short-lived connections (e.g., HTTP or RPC requests), as these connections could suffer from increased latency. Maybe something worth mentioning in your doc, as a caveat for this Alpha version.

Reply:
Yes, your understanding is correct. But I feel it's not very worth to say it's not recommended for services with short-lived connections given it's only slightly decreased, and it's hard to say whether it's overall better or worse given the subsequent packets can benefit from reduced latency. I could mention the cost of establishing a new connection is higher than NAT mode and the latency of the first packet of a connection is higher, what do you think?

Reply:
Sounds good to me. I think you have addressed this in your documentation PR.